platform/security/idp/security-idp-plugin/security-idp-plugin-nameidpolicy/src/test/groovy/org/codice/ddf/security/idp/plugin/nameidpolicy/NameIdPolicyPresignPluginSpec.groovy

```diff
@@ -0,0 +1,417 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.security.idp.plugin.nameidpolicy
+
+import org.apache.cxf.staxutils.StaxUtils
+import org.apache.wss4j.common.saml.OpenSAMLUtil
+import org.opensaml.core.xml.XMLObject
+import org.opensaml.saml.saml2.core.Attribute
+import org.opensaml.saml.saml2.core.AuthnRequest
+import org.opensaml.saml.saml2.core.NameID
+import org.opensaml.saml.saml2.core.NameIDPolicy
+import org.opensaml.saml.saml2.core.Response
+import org.w3c.dom.Document
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class NameIdPolicyPresignPluginSpec extends Specification {
+    def static INCORRECT_VALUE = "incorrect"
+    def static EXAMPLE_SP_NAME_QUALIFIER = "https://example-ddf-sp.com"
+    def static EXAMPLE_NAME_ID_VALUE = "admin"
+    def static EXAMPLE_EMAIL = "admin@localhost"
+
+    def static NAME_IDENTIFIER_URI =
+            "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"
+    def static EMAIL_URI = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
+
+    def plugin = new NameIdPolicyPresignPlugin()
+
+    Response responseMock
+    NameID nameIdMock
+    Attribute emailAttribute
+
+    def authnRequestMock = Mock(AuthnRequest)
+    def nameIdPolicyMock = Mock(NameIDPolicy)
+    def spMetadataMock = new ArrayList<>()
+    def supportedBindingsMock = new HashSet<>()
+
+    def setupSpec() {
+        OpenSAMLUtil.initSamlEngine()
+    }
+
+    def setup() {
+        def cannedResponse = this.getClass().getResource('/sampleResponse.xml').text
+        Document responseDoc = StaxUtils.read(new ByteArrayInputStream(cannedResponse.getBytes()))
+        XMLObject responseXmlObject = OpenSAMLUtil.fromDom(responseDoc.getDocumentElement())
+        responseMock = (Response) responseXmlObject
+        nameIdMock = responseMock.assertions.first().subject.nameID
+        nameIdMock.setValue(EXAMPLE_NAME_ID_VALUE)
+        nameIdMock.setFormat(NameID.UNSPECIFIED)
+
+        emailAttribute =
+                responseMock.assertions.first().attributeStatements.first().attributes.first()
+
+        authnRequestMock.getNameIDPolicy() >> nameIdPolicyMock
+    }
+
+    @Unroll("null NameIDPolicy Format tests for Format #format")
+    def 'null NameIDPolicy Format tests'() {
```

Comment:
✏️ None of these test names explain what they're doing or verifying. For example, this one could be `test that the nameID is not modified when NameIDPolicy format is null`